app/controllers/users_controller.rb

```diff
@@ -351,10 +351,14 @@ def update
 	end
 
 	def favorite_recipe
-		current_user.recipes.create(:type => "FavoritedRecipe", :meal_type => params[:Type], :title => params[:Title], 
-			:calories => params[:Calories], :time => params[:PrepTime]
-			)
-		current_user.save!
+		previously_favorited = current_user.recipes.where(:type => "FavoritedRecipe", :meal_type => params[:Type], :title => params[:Title], 
```

Comment:
I'm really glad we're doing this! It would have been really easy to overlook this one and it would have caused a ton  of pain later down the line.